README.md
A ROS rosgraph_msgs package for [dartros](https://pub.dev/packages/dartros)

CHANGELOG.md
## 0.1.0
- Stable null-safe

## 0.1.0-nullsafety.0
- Update for nullsafety

## 0.0.3
- Update to version 3 of generated messages
  
## 0.0.2
- Better typing

## 0.0.1
- Initial version, starting to expose it publicly to enable pub dependency, still a work in progress

lib/src/msgs/Log.dart
// Auto-generated. Do not edit!

// Updated: Thu Mar 18 08:38:00 2021

// (in-package rosgraph_msgs.msg)


// ignore_for_file: unused_import, overridden_fields
import 'dart:convert';
import 'package:buffer/buffer.dart';
import 'package:dartros_msgutils/msg_utils.dart';
import 'package:std_msgs/msgs.dart';

//-----------------------------------------------------------

class Log extends RosMessage<Log> {
  Header header;

  int level;

  String name;

  String msg;

  String file;

  String function;

  int line;

  List<String> topics;

  static Log $prototype = Log();
  Log({ 
    Header? header,
    int? level,
    String? name,
    String? msg,
    String? file,
    String? function,
    int? line,
    List<String>? topics,
  }):
  this.header = header ?? Header(),
  this.level = level ?? 0,
  this.name = name ?? '',
  this.msg = msg ?? '',
  this.file = file ?? '',
  this.function = function ?? '',
  this.line = line ?? 0,
  this.topics = topics ?? [];

  @override
  Log call({ 
    Header? header,
    int? level,
    String? name,
    String? msg,
    String? file,
    String? function,
    int? line,
    List<String>? topics,
  }) => Log(
  header: header,
  level: level,
  name: name,
  msg: msg,
  file: file,
  function: function,
  line: line,
  topics: topics,
  );

  void serialize(ByteDataWriter writer) {
    // Serializes a message object of type Log
    // Serialize message field [header]
    header.serialize(writer);
    // Serialize message field [level]
    writer.writeUint8(level);
    // Serialize message field [name]
    writer.writeString(name);
    // Serialize message field [msg]
    writer.writeString(msg);
    // Serialize message field [file]
    writer.writeString(file);
    // Serialize message field [function]
    writer.writeString(function);
    // Serialize message field [line]
    writer.writeUint32(line);
    // Serialize message field [topics]
    writer.writeArray<String>(topics, (val) => writer.writeString(val), specArrayLen: null);
  }

  @override
  Log deserialize(ByteDataReader reader) {
    //deserializes a message object of type Log
    final data = Log();
    // Deserialize message field [header]
    data.header = Header.$prototype.deserialize(reader);
    // Deserialize message field [level]
    data.level = reader.readUint8();
    // Deserialize message field [name]
    data.name = reader.readString();
    // Deserialize message field [msg]
    data.msg = reader.readString();
    // Deserialize message field [file]
    data.file = reader.readString();
    // Deserialize message field [function]
    data.function = reader.readString();
    // Deserialize message field [line]
    data.line = reader.readUint32();
    // Deserialize message field [topics]
    data.topics = reader.readArray<String>(() => reader.readString(), arrayLen: null);
    return data;
  }

  int getMessageSize() {
    var length = 0;
    length += header.getMessageSize();
    length += utf8.encode(name).length;
    length += utf8.encode(msg).length;
    length += utf8.encode(file).length;
    length += utf8.encode(function).length;
    topics.forEach((val) {
      length += 4 + utf8.encode(val).length;
    });
    return length + 25;
  }

  @override
  String get fullType {
    // Returns string type for a message object
    return 'rosgraph_msgs/Log';
  }

  @override
  String get md5sum {
    //Returns md5sum for a message object
    return 'acffd30cd6b6de30f120938c17c593fb';
  }

  @override
  String get messageDefinition {
    // Returns full string definition for message
    return '''##
## Severity level constants
##
byte DEBUG=1 #debug level
byte INFO=2  #general level
byte WARN=4  #warning level
byte ERROR=8 #error level
byte FATAL=16 #fatal/critical level
##
## Fields
##
Header header
byte level
string name # name of the node
string msg # message 
string file # file the message came from
string function # function the message came from
uint32 line # line the message came from
string[] topics # topic names that the node publishes

================================================================================
MSG: std_msgs/Header
# Standard metadata for higher-level stamped data types.
# This is generally used to communicate timestamped data 
# in a particular coordinate frame.
# 
# sequence ID: consecutively increasing ID 
uint32 seq
#Two-integer timestamp that is expressed as:
# * stamp.sec: seconds (stamp_secs) since epoch (in Python the variable is called 'secs')
# * stamp.nsec: nanoseconds since stamp_secs (in Python the variable is called 'nsecs')
# time-handling sugar is provided by the client library
time stamp
#Frame this data is associated with
string frame_id

''';
  }

// Constants for message
  static const int DEBUG = 1;
  static const int INFO = 2;
  static const int WARN = 4;
  static const int ERROR = 8;
  static const int FATAL = 16;

}



lib/src/msgs/Clock.dart
// Auto-generated. Do not edit!

// Updated: Thu Mar 18 08:38:00 2021

// (in-package rosgraph_msgs.msg)


// ignore_for_file: unused_import, overridden_fields
import 'dart:convert';
import 'package:buffer/buffer.dart';
import 'package:dartros_msgutils/msg_utils.dart';

//-----------------------------------------------------------

class Clock extends RosMessage<Clock> {
  RosTime clock;

  static Clock $prototype = Clock();
  Clock({ 
    RosTime? clock,
  }):
  this.clock = clock ?? RosTime(secs: 0, nsecs: 0);

  @override
  Clock call({ 
    RosTime? clock,
  }) => Clock(
  clock: clock,
  );

  void serialize(ByteDataWriter writer) {
    // Serializes a message object of type Clock
    // Serialize message field [clock]
    writer.writeTime(clock);
  }

  @override
  Clock deserialize(ByteDataReader reader) {
    //deserializes a message object of type Clock
    final data = Clock();
    // Deserialize message field [clock]
    data.clock = reader.readTime();
    return data;
  }

  int getMessageSize() {
    return 8;
  }

  @override
  String get fullType {
    // Returns string type for a message object
    return 'rosgraph_msgs/Clock';
  }

  @override
  String get md5sum {
    //Returns md5sum for a message object
    return 'a9c97c1d230cfc112e270351a944ee47';
  }

  @override
  String get messageDefinition {
    // Returns full string definition for message
    return '''# roslib/Clock is used for publishing simulated time in ROS. 
# This message simply communicates the current time.
# For more information, see http://www.ros.org/wiki/Clock
time clock

''';
  }

}



lib/src/msgs/TopicStatistics.dart
// Auto-generated. Do not edit!

// Updated: Thu Mar 18 08:38:00 2021

// (in-package rosgraph_msgs.msg)


// ignore_for_file: unused_import, overridden_fields
import 'dart:convert';
import 'package:buffer/buffer.dart';
import 'package:dartros_msgutils/msg_utils.dart';

//-----------------------------------------------------------

class TopicStatistics extends RosMessage<TopicStatistics> {
  String topic;

  String node_pub;

  String node_sub;

  RosTime window_start;

  RosTime window_stop;

  int delivered_msgs;

  int dropped_msgs;

  int traffic;

  RosTime period_mean;

  RosTime period_stddev;

  RosTime period_max;

  RosTime stamp_age_mean;

  RosTime stamp_age_stddev;

  RosTime stamp_age_max;

  static TopicStatistics $prototype = TopicStatistics();
  TopicStatistics({ 
    String? topic,
    String? node_pub,
    String? node_sub,
    RosTime? window_start,
    RosTime? window_stop,
    int? delivered_msgs,
    int? dropped_msgs,
    int? traffic,
    RosTime? period_mean,
    RosTime? period_stddev,
    RosTime? period_max,
    RosTime? stamp_age_mean,
    RosTime? stamp_age_stddev,
    RosTime? stamp_age_max,
  }):
  this.topic = topic ?? '',
  this.node_pub = node_pub ?? '',
  this.node_sub = node_sub ?? '',
  this.window_start = window_start ?? RosTime(secs: 0, nsecs: 0),
  this.window_stop = window_stop ?? RosTime(secs: 0, nsecs: 0),
  this.delivered_msgs = delivered_msgs ?? 0,
  this.dropped_msgs = dropped_msgs ?? 0,
  this.traffic = traffic ?? 0,
  this.period_mean = period_mean ?? RosTime(secs: 0, nsecs: 0),
  this.period_stddev = period_stddev ?? RosTime(secs: 0, nsecs: 0),
  this.period_max = period_max ?? RosTime(secs: 0, nsecs: 0),
  this.stamp_age_mean = stamp_age_mean ?? RosTime(secs: 0, nsecs: 0),
  this.stamp_age_stddev = stamp_age_stddev ?? RosTime(secs: 0, nsecs: 0),
  this.stamp_age_max = stamp_age_max ?? RosTime(secs: 0, nsecs: 0);

  @override
  TopicStatistics call({ 
    String? topic,
    String? node_pub,
    String? node_sub,
    RosTime? window_start,
    RosTime? window_stop,
    int? delivered_msgs,
    int? dropped_msgs,
    int? traffic,
    RosTime? period_mean,
    RosTime? period_stddev,
    RosTime? period_max,
    RosTime? stamp_age_mean,
    RosTime? stamp_age_stddev,
    RosTime? stamp_age_max,
  }) => TopicStatistics(
  topic: topic,
  node_pub: node_pub,
  node_sub: node_sub,
  window_start: window_start,
  window_stop: window_stop,
  delivered_msgs: delivered_msgs,
  dropped_msgs: dropped_msgs,
  traffic: traffic,
  period_mean: period_mean,
  period_stddev: period_stddev,
  period_max: period_max,
  stamp_age_mean: stamp_age_mean,
  stamp_age_stddev: stamp_age_stddev,
  stamp_age_max: stamp_age_max,
  );

  void serialize(ByteDataWriter writer) {
    // Serializes a message object of type TopicStatistics
    // Serialize message field [topic]
    writer.writeString(topic);
    // Serialize message field [node_pub]
    writer.writeString(node_pub);
    // Serialize message field [node_sub]
    writer.writeString(node_sub);
    // Serialize message field [window_start]
    writer.writeTime(window_start);
    // Serialize message field [window_stop]
    writer.writeTime(window_stop);
    // Serialize message field [delivered_msgs]
    writer.writeInt32(delivered_msgs);
    // Serialize message field [dropped_msgs]
    writer.writeInt32(dropped_msgs);
    // Serialize message field [traffic]
    writer.writeInt32(traffic);
    // Serialize message field [period_mean]
    writer.writeTime(period_mean);
    // Serialize message field [period_stddev]
    writer.writeTime(period_stddev);
    // Serialize message field [period_max]
    writer.writeTime(period_max);
    // Serialize message field [stamp_age_mean]
    writer.writeTime(stamp_age_mean);
    // Serialize message field [stamp_age_stddev]
    writer.writeTime(stamp_age_stddev);
    // Serialize message field [stamp_age_max]
    writer.writeTime(stamp_age_max);
  }

  @override
  TopicStatistics deserialize(ByteDataReader reader) {
    //deserializes a message object of type TopicStatistics
    final data = TopicStatistics();
    // Deserialize message field [topic]
    data.topic = reader.readString();
    // Deserialize message field [node_pub]
    data.node_pub = reader.readString();
    // Deserialize message field [node_sub]
    data.node_sub = reader.readString();
    // Deserialize message field [window_start]
    data.window_start = reader.readTime();
    // Deserialize message field [window_stop]
    data.window_stop = reader.readTime();
    // Deserialize message field [delivered_msgs]
    data.delivered_msgs = reader.readInt32();
    // Deserialize message field [dropped_msgs]
    data.dropped_msgs = reader.readInt32();
    // Deserialize message field [traffic]
    data.traffic = reader.readInt32();
    // Deserialize message field [period_mean]
    data.period_mean = reader.readTime();
    // Deserialize message field [period_stddev]
    data.period_stddev = reader.readTime();
    // Deserialize message field [period_max]
    data.period_max = reader.readTime();
    // Deserialize message field [stamp_age_mean]
    data.stamp_age_mean = reader.readTime();
    // Deserialize message field [stamp_age_stddev]
    data.stamp_age_stddev = reader.readTime();
    // Deserialize message field [stamp_age_max]
    data.stamp_age_max = reader.readTime();
    return data;
  }

  int getMessageSize() {
    var length = 0;
    length += utf8.encode(topic).length;
    length += utf8.encode(node_pub).length;
    length += utf8.encode(node_sub).length;
    return length + 88;
  }

  @override
  String get fullType {
    // Returns string type for a message object
    return 'rosgraph_msgs/TopicStatistics';
  }

  @override
  String get md5sum {
    //Returns md5sum for a message object
    return '10152ed868c5097a5e2e4a89d7daa710';
  }

  @override
  String get messageDefinition {
    // Returns full string definition for message
    return '''# name of the topic
string topic

# node id of the publisher
string node_pub

# node id of the subscriber
string node_sub

# the statistics apply to this time window
time window_start
time window_stop

# number of messages delivered during the window
int32 delivered_msgs
# numbers of messages dropped during the window
int32 dropped_msgs

# traffic during the window, in bytes
int32 traffic

# mean/stddev/max period between two messages
duration period_mean
duration period_stddev
duration period_max

# mean/stddev/max age of the message based on the
# timestamp in the message header. In case the
# message does not have a header, it will be 0.
duration stamp_age_mean
duration stamp_age_stddev
duration stamp_age_max

''';
  }

}

